plugins/modules/python_requirements_info.py

```diff
@@ -13,6 +13,10 @@
 description:
   - Get info about available Python requirements on the target host, including listing required libraries and gathering versions.
   - This module was called C(python_requirements_facts) before Ansible 2.9. The usage did not change.
+  - Although the extra identifiers are valid input, only the correctness of the identifier is verified. The library verification is not performed.
+  - This module is almost PEP-508 compliant.
+  - The requirements C(SomeProject == 5.4 ; python_version < "3.8"), C(SomeProject ; sys_platform == 'win32') are not considered valid.
+  - The version operator C(~=) is also not considered valid on every platform.
```

Comment:
Ha, I thought `pkg_resources` would be able to handle those cases.

Reply:
I thought also but it seems to be platform-dependant. Check the CI results, some are working, some are giving errors.

Also, it seems we should not be using `pkg_resources` after Python 3.8. Going to check that.

Reply:
Ok, I did not know about the platform dependency. Fair enough. Any idea on how to go around `pkg_resources` on >3.8 ?

Reply:
Yes, I further checked it but my time has been limited. According to `pkg_resources` documentation, we should be using the embedded importlib:

> Use of pkg_resources is discouraged in favor of [importlib.resources](https://docs.python.org/3/library/importlib.html#module-importlib.resources), [importlib.metadata](https://docs.python.org/3/library/importlib.metadata.html), and their backports ([importlib_resources](https://pypi.org/project/importlib_resources), [importlib_metadata](https://pypi.org/project/importlib_metadata)). Please consider using those libraries instead of pkg_resources.

The migration guide being here: https://importlib-metadata.readthedocs.io/en/latest/migration.html

I am only worried that this could introduce a difference in how the module behaves (especially in term of input validation) between the two libs. Also the support for importlib is not the same between 3.8 and 3.9.

A third option that I have found, but not ideal because it adds yet another library is to use this pep-508 package to parse the input: https://pypi.org/project/pep508/

This could streamline it.

Definitely, the tag `easyfix` on the issue isn't or we need to scope exactly what this PR should support :-)